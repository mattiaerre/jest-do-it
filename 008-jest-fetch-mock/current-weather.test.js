const fetch = require('jest-fetch-mock');

jest.setMock('node-fetch', fetch);
const factory = require('./current-weather');

describe('current-weather', () => {
  const config = { appid: 'appid' };
  const currentWeather = factory(config);

  test('currentWeather()', async () => {
    const weather = { temperature: '25.17 Â°C' };
    fetch.mockResponse(JSON.stringify(weather));

    const current = await currentWeather({ city: 'London' });

    expect(current).toEqual(weather);
    expect(current).toMatchSnapshot();
  });

  test('fetch', () => {
    expect(fetch).toHaveBeenCalled();
    expect(fetch.mock.calls).toMatchSnapshot();
  });
});
