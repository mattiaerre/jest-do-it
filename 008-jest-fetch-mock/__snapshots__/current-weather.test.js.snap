// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`current-weather currentWeather() 1`] = `
Object {
  "temperature": "25.17 Â°C",
}
`;

exports[`current-weather fetch 1`] = `
Array [
  Array [
    "http://api.openweathermap.org/data/2.5/weather?q=London&appid=appid&units=metric",
  ],
]
`;
